src/protocol/mod.rs

```diff
@@ -0,0 +1,53 @@
+use crate::packet::{HandshakeControlInfo, ShakeType};
+use crate::ControlPacket;
+use std::net::SocketAddr;
+use std::time::Instant;
+
+pub mod sender;
+
+pub enum Handshake {
+    Caller,
```

Comment:
Again I was experimenting with names a bit to provoke discussion. I also needed something to use as the top level API to drive a test. I hadn't settled on a name yet and went with caller because Caller-Listener is what the spec uses in this scenario. Upon further thought I actually think ultimately where I would suggest we go is to mirror the TCP implementation in tokio, it has TcpListener and TcpStream. So we'd have a SrtListener and SrtStream. SrtListener would be the only way to start a listener connection. It would have an ccept() and incoming() implementation that allow provide an SrtStream for each new connection, just like TcpListener, and SrtStream would have two connection methods: call() and rendezvous(). Would that work for you?